### PR TITLE
[JENKINS-54460] - Multi-architecture packaging for jenkins/slave

### PR DESCRIPTION
**Pull request by oleg-nenashev**
It inherits https://github.com/jenkinsci/docker/pull/719 from @jmreicha and demonstrates multi-platform Docker image packaging for Agent images. It also adds multi-java-version support to the script.

CC @jenkinsci/sig-platform

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
Generally it should be a part of https://issues.jenkins-ci.org/browse/JENKINS-54460

File: build-docker.sh [hunk 1]
```diff
@@ -0,0 +1,433 @@
+#!/bin/bash -eu
```
Comment (darxriggs) [inline build-docker.sh]: `-eu` is duplicated here and should be removed as it is already set in line 8.

File: multiarch/Dockerfile.alpine [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ENV HOME /home/${user}
+RUN addgroup -g ${gid} ${group}
+RUN adduser -h $HOME -u ${uid} -G ${group} -D ${user}
+LABEL Description="This is a base image, which provides the Jenkins agent executable (slave.jar)" Vendor="Jenkins project" Version="3.23"
+
+ARG REMOTING_VERSION=3.26
+ARG AGENT_WORKDIR=/home/${user}/agent
+
+RUN apk add --update --no-cache curl bash git openssh-client openssl procps \
+  && curl --create-dirs -sSLo /usr/share/jenkins/slave.jar https://repo.jenkins-ci.org/public/org/jenkins-ci/main/remoting/${REMOTING_VERSION}/remoting-${REMOTING_VERSION}.jar \
```
Comment (darxriggs) [inline multiarch/Dockerfile.alpine]: Wasn't there a renaming away from "slave"?

**Comment by timja** [comment]
I think this can be closed, isn't this the result of this work? https://github.com/jenkinsci/docker-agent

**Pull request closed without merging**